pkg/envinfo/types.go

```diff
@@ -0,0 +1,76 @@
+package envinfo
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+	"text/tabwriter"
+
+	"github.com/openshift/odo/pkg/localConfigProvider"
+	"github.com/openshift/odo/pkg/machineoutput"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Info struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ComponentSettings `json:"spec"`
+}
+
+const InfoKind = "EnvInfo"
+
+// ComponentSettings holds all component related information
+type ComponentSettings struct {
+	Name string `yaml:"Name,omitempty" json:"name,omitempty"`
+
+	Project string `yaml:"Project,omitempty" json:"project,omitempty"`
+
+	UserCreatedDevfile bool `yaml:"UserCreatedDevfile,omitempty" json:"UserCreatedDevfile,omitempty"`
+
+	URL *[]localConfigProvider.LocalURL `yaml:"Url,omitempty" json:"url,omitempty"`
+	// AppName is the application name. Application is a virtual concept present in odo used
+	// for grouping of components. A namespace can contain multiple applications
+	AppName string `yaml:"AppName,omitempty" json:"appName,omitempty"`
+
+	// DebugPort controls the port used by the pod to run the debugging agent on
+	DebugPort *int `yaml:"DebugPort,omitempty" json:"debugPort,omitempty"`
```

Comment:
I understand that a pointer is used here because the DebugPort value can be non accurate in some circumstances. And in these circumstances, the value of DebugPort will be `nil`. In the other cases, `*DebutPort` will contain the port value.

Reply:
I can't imagine what circumstances this would be. When I see an `*int` I always think it's difficult to work with than an `int` because you have to make sure to first assign it to a variable and then reference it.